echopop/tests/test_transect_functions.py

```diff
@@ -0,0 +1,311 @@
+    test_acoustics_dict = copy.deepcopy( objS.acoustics )
+    test_biology_dict = copy.deepcopy( objS.biology ) 
+    test_info_strata = objS.spatial[ 'strata_df' ].copy( )
+
+    ### Evaluate object for later comparison 
+    eval_nasc_fraction_total_df = index_transect_age_sex_proportions( test_acoustics_dict ,
+                                                                      test_biology_dict , 
+                                                                      test_info_strata )
+
+    ###--------------------------------
+    ### Expected outcomes
+    ###--------------------------------
+    # ---- Expected dimensions
+    expected_dimensions = tuple( [ 8 , 24 ] )
+    # ---- Expected output
+    expected_output = pd.DataFrame(
+        {
+            'latitude': np.repeat( [ 20.0 , 30.0 , 40.0 , 50.0 ] , 2 ) ,
+            'longitude': np.repeat( [ -180.0 , -120.0 , -170.0 , -110.0 ] , 2 ) ,
+            'transect_num': np.repeat( [ 1 , 2 , 3 , 4 ] , 2 ).astype( np.int64 ) ,
+            'stratum_num': np.repeat( [ 0 , 1 ] , 4 ).astype( np.int64 ) ,
+            'haul_num': np.repeat( [ 1 , 2 ] , 4 ).astype( np.int64 ) ,
+            'interval': np.repeat( [ 10.0 , 10.0 , 10.0 , 9.9 ] , 2 ) ,
+            'interval_area': np.repeat( [ 10.0 , 10.0 , 10.0 , 9.9 ] , 2 ) ,
+            'NASC_all_ages': np.repeat( [ 1e1 , 1e2 , 1e2 , 1e3 ] , 2 ) ,
+            'NASC_no_age1': np.repeat( [ 0.0 , 1e1 , 1e2 , 1e3 ] , 2 ) ,
+            'fraction_hake': np.repeat( [ 1.0 , 0.5 ] , 4 ) ,
+            'species_id': np.repeat( 8675309 , 8 ).astype( np.int64 ) ,
+            'sigma_bs_mean': np.repeat( 1.630277e-8 , 8 ) ,
+            'proportion_female': np.repeat( [ 0.592593 , 0.407407 ] , 4 ) ,
+            'proportion_male': np.repeat( [ 0.407407 , 0.592593 ] , 4 ) ,
+            'proportion_station_1': np.repeat( 0.925926 , 8 ) ,
+            'proportion_station_2': np.repeat( 0.074074 , 8 ) ,
+            'average_weight_female': np.repeat( [ 4.719110 , 2.707892 ] , 4 ) ,
+            'average_weight_male': np.repeat( [ 6.640487 , 6.299942 ] , 4 ) ,
+            'average_weight_total': np.repeat( [ 3.066481 , 2.603519 ] , 4 ) ,
+            'age': np.tile( [ 1 , 2 ] , 4 ).astype( np.int64 ) ,
+            'count_age_proportion_all': np.repeat( 0.5 , 8 ) ,
+            'count_age_proportion_adult': np.tile( [ 0.0 , 1.0 ] , 4 ) ,
+            'weight_age_proportion_all': np.repeat( 0.5 , 8 ) ,
+            'weight_age_proportion_adult': np.tile( [ 0.0 , 1.0 ] , 4 )
+        }
+    )
+
+    #----------------------------------
+    ### Run tests: `index_transect_age_sex_proportions`
+    #----------------------------------
+    ### Check shape 
+    assert eval_nasc_fraction_total_df.shape == expected_dimensions
+    ### Check datatypes
+    assert np.all( eval_nasc_fraction_total_df.dtypes == expected_output.dtypes )
+    ### Dataframe equality
+    assert np.allclose( eval_nasc_fraction_total_df , expected_output , rtol = 1e-1 )
```

Comment:
So the function I was telling you about was [`np.issubdtype`](https://numpy.org/doc/stable/reference/generated/numpy.issubdtype.html#numpy.issubdtype):
```python
# below will both pass
assert np.issubdtype(np.int32, np.integer)
assert np.issubdtype(np.int64, np.integer)
```

For the tests here, I think you can specify separate dicts for the expected dtype and expected values, like how you separate out the expected dimensions, and test the dtype and values separately:
```python
expected_dtype = {
    'longitude': np.floating,
    'transect_num': np.integer,
    'stratum_num': np.integer,
     ...
}
expected_values = {
    'longitude': np.repeat( [ -180.0 , -120.0 , -170.0 , -110.0 ] , 2 ) ,
    'transect_num': np.repeat( [ 1 , 2 , 3 , 4 ] , 2 ),
    'stratum_num': np.repeat( [ 0 , 1 ] , 4 ),
    ...
}

# test each column separately
for col in eval_nasc_fraction_total_df.columns:
    assert np.issubdtype(eval_nasc_fraction_total_df[col], expected_dtype[col])
    assert np.allclose(eval_nasc_fraction_total_df[col].data, expected_values[col])

```

You may need to use `np.allclose` to test for equality for `np.floating` types. And just a small note on what we discussed that rtol=1e-1 is a pretty big tolerance, and may signal that something may be going on underneath.